lib/oidcstrategy.js

```diff
@@ -342,33 +350,15 @@ Strategy.prototype.authenticate = function authenticateStrategy(req, options) {
 
                   // use try / catch around JSON.parse --> could fail unexpectedly
                   try {
-                    return onProfileLoaded(self, {
-                      req,
-                      sub,
-                      iss,
-                      profile: makeProfileObject(JSON.parse(body), body),
-                      jwtClaims,
-                      accessToken,
-                      refreshToken,
-                      params,
-                    });
+                    // replace profile with the userinfo
+                    profile = makeProfileObject(JSON.parse(body), body);
                   } catch (ex) {
                     return next(ex);
                   }
                 });
               }
-
-              // lets do an id_token fallback. We use id_token over userInfo endpoint for now
-              return onProfileLoaded(self, {
-                req,
-                sub,
-                iss,
-                profile: (idToken) ? makeProfileObject(jwtClaims, jwtClaimsStr) : undefined,
-                jwtClaims,
-                accessToken,
-                refreshToken,
-                params,
-              });
+              
+              return onProfileLoaded(self, {req, sub, iss, profile, jwtClaims, accessToken, refreshToken, params});
```

Comment:
if `_shouldLoadUserProfile` resolves `load === true`, this code tries to load a user profile from `userInfoURL` and `makeProfileObject` parses the response body of that call.

otherwise, `makeProfileObject` parses the previously retrieved `jwtClaims`

IMHO, this logic can be removed since the v2.0 endpoints don't provide a user info endpoint and thus, profile is always parsed from `jwtClaims`